### DISCUSSION TITLE
FastAPI shows shutting down while inferencing using pytorch models

### DISCUSSION THREAD

**Discussion by pradeepdev-1995** [fastapi/fastapi#8872] [Questions] [answered]
I tried to load and inference from a pytorch model using fastAPI(with uvicorn).
But while predicting time it shows like this

INFO: Shutting down
INFO: Waiting for connections to close. (CTRL+C to force quit)

I tried the same method using flask and worked fine within milliseconds.

Here is the code for pretrained model prediction using fastapi

![Screenshot from 2021-05-24 10-28-39](https://user-images.githubusercontent.com/41164884/119298771-4cc65e00-bc7b-11eb-98d2-8280cb84d1a6.png)

But While predicting with fastAPI it shows like this

![Screenshot from 2021-05-24 09-53-15](https://user-images.githubusercontent.com/41164884/119298791-5780f300-bc7b-11eb-9b6e-b530d92a330b.png)

(Since it is a pytorch RoBERTA based huge pretrained model, the model size is around 600MB)

**Comment by ycd** [comment]
I can't see what is the issue in here?

**Comment by pradeepdev-1995** [comment]
@ycd During the inference time it shows like this

**INFO: Shutting down
INFO: Waiting for connections to close. (CTRL+C to force quit)**                                           

so model prediction not happening in fastAPI(same method working fine with flask)

**Comment by ycd** [comment]
FastAPI has no idea about the running process and how you are running it, check out your ASGI server again.

**Comment by pradeepdev-1995** [comment]
@ycd  okay. So could you tell me how can I check the ASGI?

**Comment by santosh-gokul** [comment, marked as answer]
HI @pradeepdev-1995 , I've managed to replicate the issue that you are facing, it looks like the Uvicorn gets Signal#15, which makes it close the server.

Try this  `ClassificationModel('roberta','outputs', num_labels = total_classes,use_cuda=False, args = {"use_multiprocessing":False})`, it is good if it works, else it looks like it has no immediate fix as of now, you can have a look at an interesting discussion here: 
* https://github.com/cortexlabs/cortex/issues/1346
* https://github.com/ThilinaRajapakse/simpletransformers/issues/705

**Comment by pradeepdev-1995** [comment]
@santosh-gokul  Thanks a lot. The solution worked for me.
